src/check.rs

```diff
@@ -94,42 +179,183 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
         }
     }
 
-    let resp = ureq::head(url.as_str()).call();
-    if resp.synthetic() {
-        Err(CheckError::Http(Box::new(HttpError::Fetch(
-            url.clone(),
-            resp.into_synthetic_error().unwrap(),
-        ))))
-    } else if resp.ok() {
-        Ok(())
+    // only if the URL contains a fragment we need to fetch the body via
+    // GET, otherwise we can use HEAD
+    if url.fragment().is_none() {
+        let resp = ureq::head(url.as_str()).call();
+
+        if resp.synthetic() {
+            Err(CheckError::Http(Box::new(HttpError::Fetch(
+                url.clone(),
+                resp.into_synthetic_error().unwrap(),
+            ))))
+        } else if resp.ok() {
+            Ok(())
+        } else {
+            Err(CheckError::Http(Box::new(HttpError::UnexpectedStatus(
+                url.clone(),
+                resp,
+            ))))
+        }
```

Comment:
I'd turn this into a free function taking a `Response` and returning a `Result<(), CheckError>`. It shouldn't be necessary in ureq 2.0 (https://github.com/algesten/ureq/pull/132) but that's still a ways away.